addons/web_editor/static/src/js/wysiwyg/widgets/media.js

```diff
@@ -277,14 +278,17 @@ var ImageWidget = MediaWidget.extend({
             domain = domain.concat(attachedDocumentDomain);
         }
         domain = ['|', ['public', '=', true]].concat(domain);
-
         domain.push('|',
             ['mimetype', '=', false],
             ['mimetype', this.options.document ? 'not in' : 'in', ['image/gif', 'image/jpe', 'image/jpeg', 'image/jpg', 'image/gif', 'image/png']]);
         if (needle && needle.length) {
             domain.push('|', ['datas_fname', 'ilike', needle], ['name', 'ilike', needle]);
         }
         domain.push('|', ['datas_fname', '=', false], '!', ['datas_fname', '=like', '%.crop'], '!', ['name', '=like', '%.crop']);
+        // the assets should not be part of the documents
+        if (this.options.document) {
+            domain = domain.concat('!', '&', utils.assetsDomain());
```

Comment:
Creating domains in JS is bad I know :) Let's put the `&` inside the `assetsDomain` implementation in utils.js so that this domain can more easily be combined with other domains.